### PR TITLE
Added get_names to Extractors.

### PR DESCRIPTION
**Pull request by luca-heltai**
Extrapolate from #7806 extractor names.

### CODE REVIEW

File: include/deal.II/fe/fe_values_extractors.h [hunk 1]
```diff
@@ -112,6 +114,12 @@ namespace FEValuesExtractors
      * Constructor. Take the selected vector component as argument.
      */
     Scalar(const unsigned int component);
+
+    /**
+     * Return a string that uniquely identifies this finite element Extractor.
```
Comment (jppelteret) [inline include/deal.II/fe/fe_values_extractors.h]: `Extractor` -> `extractor`.

Both here and below.

File: doc/news/changes/minor/20190319LucaHeltai [hunk 2]
```diff
@@ -0,0 +1,5 @@
+New: FEValuesExtractor classes now have a new method get_name() that return a unique string identifier
```
Comment (bangerth) [inline doc/news/changes/minor/20190319LucaHeltai]:
```suggestion
New: FEValuesExtractor classes now have a new method get_name() that returns a unique string identifier
```

File: include/deal.II/fe/fe_values_extractors.h [hunk 3]
```diff
@@ -301,6 +327,38 @@ namespace FEValuesExtractors
   inline Tensor<rank>::Tensor(const unsigned int first_tensor_component)
     : first_tensor_component(first_tensor_component)
   {}
+
+
+  inline std::string
+  Scalar::get_name() const
```
Comment (bangerth) [inline include/deal.II/fe/fe_values_extractors.h]: Not wrong, but could probably also be in the `.cc` file to make compiling everything else faster. I suspect that these functions aren't performance critical.

File: source/fe/fe_values_extractors.cc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  template <int rank>
+  std::string
+  SymmetricTensor<rank>::get_name() const
+  {
+    return "SymmetricTensor<" + Utilities::int_to_string(rank) + ">(" +
+           Utilities::int_to_string(first_tensor_component) + ")";
+  }
+
+  // Explicit instantiations
+  template struct Tensor<1>;
```
Comment (jppelteret) [inline source/fe/fe_values_extractors.cc]: I think that this needs to be instantiated for `Tensor<0>` as well.

**Pull request merged**